mfbt/Assertions.h

```diff
@@ -50,7 +50,21 @@ MOZ_BEGIN_EXTERN_C
 extern MFBT_DATA const char* gMozCrashReason;
 MOZ_END_EXTERN_C
 
-#if defined(MOZ_HAS_MOZGLUE) || defined(MOZILLA_INTERNAL_API)
```

Comment:
Probably we should branch on this condition.

Reply:
I'll test and merge when I have some confident on this. Those conditions not intuitive, so I'm looking around codebase.

Reply:
The former seems to all final executables and dynamic libraries to be installed
- https://github.com/kenz-gelsoft/gecko-dev/issues/97#issuecomment-2560594280

Reply:
The latter is for dom/media code and libxul.so
- https://github.com/kenz-gelsoft/gecko-dev/issues/97#issuecomment-2561469184

Reply:
We should move `#ifdef __HAIKU__` into `#if defined(MOZ_HAS_MOZGLUE) || defined(MOZILLA_INTERNAL_API)`.

Because `#else` was intentionally doing nothing. From above investigations almost all final binaries are covered with `#if defined(MOZ_HAS_MOZGLUE) || defined(MOZILLA_INTERNAL_API)` case.

Reply:
Fixed on 6128d2f332a461b4aef7fadc7da9b27a0a7159e6, and it builds.